IAsyncEnumerableExtensions.cs

```diff
@@ -38,5 +255,138 @@ public static class IAsyncEnumerableExtensions
+        /// <param name="token">A <see cref="CancellationToken"/> that can hault enumeration of <paramref name="source"/></param>
+        public static IAsyncEnumerable<TSource> SkipWhileAsync<TSource>(
+            this IAsyncEnumerable<TSource> source,
+            Func<TSource, bool> predicate,
+            bool oneTimeUse = false,
+            CancellationToken token = default(CancellationToken))
+        {
+            if (null == source)
+                throw new ArgumentNullException(nameof(source));
+            if (null == predicate)
+                throw new ArgumentNullException(nameof(predicate));
+
+            return new AsyncEnumerable<TSource>(
+                async yield =>
+                {
+                    var yielding = false;
+
+                    using (var enumerator = await source.GetAsyncEnumeratorAsync(token).ConfigureAwait(false))
+                        while (await enumerator.MoveNextAsync(token).ConfigureAwait(false))
+                        {
+                            if (!yielding && !predicate(enumerator.Current))
+                                yielding = true;
+
+                            if (yielding)
+                                await yield.ReturnAsync(enumerator.Current).ConfigureAwait(false);
+                        }
+                },
+                oneTimeUse);
+        }
+
+        #endregion
+
+        #region Where
+
+        /// <summary>
+        /// Filters a sequence of values based on a predicate.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of <paramref name="source"/></typeparam>
+        /// <param name="source">An <see cref="IAsyncEnumerable{T}"/> to filter.</param>
+        /// <param name="predicate">A function to test each element for a condition.</param>
+        /// <param name="oneTimeUse">When <c>true</c> the enumeration can be performed once only and <see cref="AsyncEnumerator{T}.Reset"/> method is not allowed</param>
+        /// <param name="token">A <see cref="CancellationToken"/> that can hault enumeration of <paramref name="source"/></param>
+        public static IAsyncEnumerable<TSource> WhereAsync<TSource>(
+            this IAsyncEnumerable<TSource> source,
+            Func<TSource, bool> predicate,
+            bool oneTimeUse = false,
+            CancellationToken token = default(CancellationToken))
+        {
+            if (null == source)
+                throw new ArgumentNullException(nameof(source));
+            if (null == predicate)
+                throw new ArgumentNullException(nameof(predicate));
+
+            return new AsyncEnumerable<TSource>(
+                yield =>
+                    source.ForEachAsync(async item =>
+                    {
+                        if (predicate(item))
+                            await yield.ReturnAsync(item).ConfigureAwait(false);
```

Comment:
No need for ```async```:
     if (predicate(item))
         return yield.ReturnAsync(item);
    else
        return TaskEx.Completed;

p.s. please add ```TaskEx```